pd/src/state.rs

```diff
@@ -27,6 +24,12 @@ pub struct State {
     pool: Pool<Postgres>,
 }
 
+// NOTE: because the state has a Postgres connection pool and supports shared
+// access, all methods (including write methods!) take &self, not &mut self.
+//
+// Writes to the database should *only* happen in `commit_genesis` (called once
+// on init) and `commit_block`.
```

Comment:
The previous `State` code had multiple methods for writing data (as an artefact of those parts being written in parallel); along the way to these changes, those methods are now combined into two `commit` methods.  Going forward, we should add new DB writes only in those places, which correspond to the ABCI state changes.